### PR TITLE
Perhaps consider going the ES6 route.

### PR DESCRIPTION
**Pull request by SaulDoesCode**
I converted this awesome animation library to es6 code added and custom build system it works exactly the same.
##### benefits 
* es6 code
* less lines of code
* build system - minifies and transpiles to es5

maybe look over make some suggestions.
say if it's bad or if you like the concept but perhaps change it your way.

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        compilation_level: 'SIMPLE',
+        language_in: 'ECMASCRIPT5_STRICT',
+        language_out: 'ECMASCRIPT5_STRICT',
+        js_output_file: './dist/polyfills.min.js'
+    }).run(() => {
+        console.log('Success -> Polyfills Bundled and Minified!\n');
+    });
+}
+
+function BabelizeBeatifyMinify(inloc, outloc) {
```
Comment (Maximilianos) [inline build.js]: should be BabelBea**u**tifyMinify :)

**Comment by SaulDoesCode** [comment]
There still seems to be one or two bugs I'm working out with the conversion. I'm testing it on your Code Pen demos. thanks for looking at it will apply your suggestion for the build.js file. will keep you informed on whats happening. still working out some kinks

File: package.json [hunk 2]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "animejs",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "description": "Polyvalent Javascript animation engine",
   "main": "anime.min.js",
   "repository": {
     "type": "git",
-    "url": "https://github.com/juliangarnier/anime.git"
+    "url": "https://github.com/sauldoescode/anime.git"
```
Comment (Maximilianos) [inline package.json]: prob want to keep julian garnier here :) and same below

Comment (SaulDoesCode) [inline package.json]: yeah sorry lol not used to working with git.

Comment (Maximilianos) [inline package.json]: no worries :)

**Comment by SaulDoesCode** [comment]
fixed silly bug with toArray, all the demos are working in ES6 version so far.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#Folders
+bower_components
+node_modules
+
+#FileTypes
+*.exe
```
Comment (tbranyen) [inline .gitignore]: Why do you need to ignore Windows executables in this library?

Comment (SaulDoesCode) [inline .gitignore]: Just copied it from one of my projects that uses .exe files. I'll remove it

**Comment by SaulDoesCode** [comment]
Perhaps users may benefit from exposing some internal utility functions such as includes, removeArrayDuplicates, flattenArray like they do with anime.random.

**Comment by lukeed** [comment]
I'd prefer **not** to see this happen 👎 

There's nothing wrong with the current distribution, and adding in ES6 only adds build complexity & usually transpilation "bloat". Your branch doesn't split the code into files, thus `require`, `import`, and `export` aren't used, but once a codebase is written in ES6, that's generally the next step.

---

Sidenote: @SaulDoesCode future note, your `package.json` should be declaring `devDependencies`, not `dependencies`.

**Comment by SaulDoesCode** [comment]
Thanks, I understand. I'll try to keep this branch updated and in sync with the main repo for whom ever it matters. Will change that in the package.json thanks.

File: anime.js [hunk 4]
```diff
@@ -1,605 +1,555 @@
...
-      anim.running = true;
-      time.start = +new Date();
-      time.last = anim.ended ? 0 : anim.time;
-      var s = anim.settings;
-      if (s.direction === 'reverse') reverseTweens(anim);
-      if (s.direction === 'alternate' && !s.loop) s.loop = 1;
-      setWillChange(anim);
-      animations.push(anim);
-      time.raf = requestAnimationFrame(time.tick);
+    typeof define === 'function' && define.amd ? define([], factory) :
```
Comment (lukeed) [inline anime.js]: incorrect ternary declarations. all these will run

File: anime.js [hunk 5]
```diff
@@ -1,605 +1,555 @@
...
+        // only CommonJS-like environments that support module.exports,
+        // like Node.
+        typeof module === 'object' && module.exports ? module.exports = factory() :
+        // Browser globals (root is window)
+        root.anime = factory();
+}(this, () => {
+    // Defaults
+    const undef = void 0;
+
+    let defaultSettings = {
```
Comment (lukeed) [inline anime.js]: use const, doesnt change

File: anime.js [hunk 6]
```diff
@@ -1,605 +1,555 @@
...
-      for (var t = animation.tweens.length-1; t >= 0; t--) {
-        var tween = animation.tweens[t];
-        for (var a = tween.animatables.length-1; a >= 0; a--) {
-          if (arrayContains(targets, tween.animatables[a].target)) {
-            tween.animatables.splice(a, 1);
-            if (!tween.animatables.length) animation.tweens.splice(t, 1);
-            if (!animation.tweens.length) animation.pause();
-          }
+    const is = (() => ({
+        array: Array.isArray,
```
Comment (lukeed) [inline anime.js]: should be a function of `a`

File: anime.js (hunk 4 above)
Comment (SaulDoesCode) [inline anime.js]: thanks for noticing will change that.

File: anime.js [hunk 7]
```diff
@@ -1,605 +1,555 @@
...
+            update: undef,
+            complete: undef
+        },
+        validTransforms = ['translateX', 'translateY', 'translateZ', 'rotate', 'rotateX', 'rotateY', 'rotateZ', 'scale', 'scaleX', 'scaleY', 'scaleZ', 'skewX', 'skewY'];
+
+    // Utils
+
+    function includes(arr, searchElement) {
+        if(arr.includes) return arr.includes(searchElement);
+        if (!Array.isArray(arr)) arr = Array.prototype.slice.call(arr);
```
Comment (lukeed) [inline anime.js]: might as well use the `is` util: 

```js
if (!is.array(arr)) //..
```

File: anime.js [hunk 8]
```diff
@@ -1,605 +1,555 @@
...
+        return arr.length == 0 ? false : arr.some(item => item === searchitem);
```
Comment (lukeed) [inline anime.js]: triple `===`

File: anime.js [hunk 9]
```diff
@@ -1,605 +1,555 @@
...
+        rgb: a => /^rgb/.test(a),
+        rgba: a => /^rgba/.test(a),
+        hsl: a => /^hsl/.test(a),
+        color: a => (is.hex(a) || is.rgb(a) || is.rgba(a) || is.hsl(a))
+    }))();
+
+    // Easings functions adapted from http://jqueryui.com/
+
+    const easings = (() => {
+        let eases = {},
```
Comment (lukeed) [inline anime.js]: split these variables, some are constants

**Comment by SaulDoesCode** [comment]
Awesome constructive criticism @lukeed thanks I'll see to it that those changes be implemented

File: anime.js [hunk 10]
```diff
@@ -1,605 +1,555 @@
...
+            eases['easeOut' + name] = (t, m) => 1 - easeIn(1 - t, m);
+            eases['easeInOut' + name] = (t, m) => t < 0.5 ? easeIn(t * 2, m) / 2 : 1 - easeIn(t * -2 + 2, m) / 2;
+        });
+        eases.linear = t => t;
+        return eases;
+    })();
+
+    // Strings
+
+    const numberToString = val => (is.string(val)) ? val : val + '',
```
Comment (lukeed) [inline anime.js]: template string: wrap `${val}` in "`"

File: anime.js (hunk 6 above)
Comment (SaulDoesCode) [inline anime.js]: it is wrapped in a function in the original repo but it doesn't change the behavior this way Array.isArray should be fine.

File: anime.js [hunk 11]
```diff
@@ -1,605 +1,555 @@
...
+
+        // Arrays
+        toArray = o => {
+            if (is.array(o)) return o;
+            if (is.string(o)) o = selectString(o) || o;
+            if (is.html(o)) return [].slice.call(o);
+            return [o];
+        },
+
+        flattenArray = arr => Array.prototype.reduce.call(arr, (a, b) => a.concat(is.array(b) ? flattenArray(b) : b), []),
```
Comment (lukeed) [inline anime.js]: leave as `arr.reduce`

File: anime.js [hunk 12]
```diff
@@ -1,605 +1,555 @@
...
+
+        groupArrayByProps = (arr, propsArr) => {
+            let groups = {};
+            arr.forEach(o => {
+                let group = JSON.stringify(propsArr.map(p => o[p]));
```
Comment (lukeed) [inline anime.js]: constant

File: anime.js [hunk 13]
```diff
@@ -1,605 +1,555 @@
...
+
+        mergeObjects = (o1, o2) => {
+            for (let p in o2) o1[p] = !is.undef(o1[p]) ? o1[p] : o2[p];
+            return o1;
+        },
+
+        // Colors
+        hexToRgb = hex => {
+            hex = hex.replace(/^#?([a-f\d])([a-f\d])([a-f\d])$/i, (m, r, g, b) => r + r + g + g + b + b);
+            let rgb = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex),
```
Comment (lukeed) [inline anime.js]: constants

File: anime.js [hunk 14]
```diff
@@ -1,605 +1,555 @@
...
+                r = parseInt(rgb[1], 16),
+                g = parseInt(rgb[2], 16),
+                b = parseInt(rgb[3], 16);
+            return 'rgb(' + r + ',' + g + ',' + b + ')';
```
Comment (lukeed) [inline anime.js]: es6 template string

File: anime.js (hunk 11 above)
Comment (SaulDoesCode) [inline anime.js]: flattenArray doesn't always receive pure arrays all the time so it makes errors this way it works with arraylike objects

**Comment by lukeed** [comment]
@SaulDoesCode There's more that I can add, but you'd receive the same critiques from [XO](https://github.com/sindresorhus/xo). You might want to check that out instead :)

**Comment by SaulDoesCode** [comment]
Very well I'll see how that linter works and follow it's guide lines. again many thanks for helping

File: anime.js (hunk 4 above)
Comment (lukeed) [inline anime.js]: It'd be a multi-nested ternary expression, which is hard to follow. And you can't use a `return` for relief either.. so just copy the original `if/elif/else` block; it's common across most javascript packages.

**Comment by lukeed** [comment]
@SaulDoesCode No problem 👌 Ping me if you need help setting up XO. It's a great tool

**Comment by SaulDoesCode** [comment]
Keeping up with all the changes so far, implemented latest from anime v1.1.0 and implemented @devmario's chainable suggestions.
nice work with the engine object and usage of performance.now so much better now. :+1:

File: anime.js [hunk 15]
```diff
@@ -534,14 +535,27 @@
...
             anim.update = callbacks('update');
             anim.complete = callbacks('complete');
 
+            const promise = type => {
+                if (typeof Promise !== "undefined" && Promise.toString().indexOf("[native code]") !== -1) {
+                    return () => new Promise(resolve => {
+                      anim[type + '_resolve'] = resolve;
+                    });
+                }
+                console.warn('Your browser doesn\'t support promises.');
```
Comment (devmario) [inline anime.js]: This warning msg sounds good more then before!
